java/main/src/main/java/com/saucelabs/saucebindings/GitManager.java

```diff
@@ -0,0 +1,24 @@
+package com.saucelabs.saucebindings;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class GitManager {
+    private static String currentBranch = "_default_";
+
+    public static String getCurrentBranch() {
+        try {
+            Process process = Runtime.getRuntime().exec("git rev-parse --abbrev-ref HEAD");
+            process.waitFor();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line = reader.readLine();
+            if (!line.contains("fatal:")) {
+                currentBranch = line;
+            }
+        } catch (IOException | InterruptedException e) {
+            // Ignore exception and use default
+        }
+        return currentBranch;
```

Comment:
is it ok that a previously remembered value is returned if an exception happens?

Reply:
Trying to think of a scenario where that could happen and be bad. It's going to just use the default value set at the top. I made these static methods/fields because no one should be switching git branches in the middle of their test execution.

Reply:
I don't think you can programmatically change branches in the middle of a test run. If it was found before, use that, if it wasn't found before, use default. This is a weird nice-to-have-but-not-very-important feature in the scheme of this project.

Reply:
this totally makes sense to me. What I want to say is that this method is going to still spend time for the command line execution even though the branch name has been already retrieved. the condition I was talking about:

```
if (!Objects.equals(currentBranch, DEFAULT_BRANCH)) {
  return currentBranch;
}

... do existing stuff to get the branch name from git
```